[11.0] [IMP] Update account_financial_report

feketemihai opened the pull request:
Update links in report, add account group file, update trial balance with hierarchy.

Commit:
Update links in report, add account group file, update trial balance with hierarchy.

Update indentation, remove empty lines from header.

Update test.

Update pylint.

Remove company_id on computing accounts, since account.group is not a company based model, filtering accounts is done on trial balance report.

Update account variables.

Improve condition in padding on accounts.

Add option to print hierarchy based on defined accounts/computed accounts.

Add VAT report, hierarchy from tax tags ans taxes.

Fix pylint, xlsx report generation header.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Update code to select code_prefix or name.

Fix domain in base amounts in vat report.

Change trial balance code_prefix or name.

Update trail balance, add tests for vat report.

Update pylint, amounts as monetary, many2one option on generation excels.

Update pulint.

Add VAT Report in readme.

Add VAT Report in readme.

Update array_agg.

Update array_agg.

Update array_agg.

Add option in VAT Report to be printed on Tax Tags - Tax Groups.

Add widget to hierarchy_on on trial balance.

JordiBForgeFlow commented:
Nice... Thanks! Will review asap!

Cc @rgarnau @etobella

feketemihai commented:
@jbeficent Still have to adjust tests.

JordiBForgeFlow commented on account_financial_report/report/templates/trial_balance.xml:
```diff
@@ -27,40 +27,123 @@
+                                <t t-set="style" t-value="'font-size: 9px; font-weight: bold; color: blue;'"/>
+                                <t t-set="padding" t-value="12"/>
+                            </t>
+                            <t t-if="line.account_group_id and len(line.code) == 5">
```
I suggest that instead of hardcoding we set a account.financial.report.style model that allows to define the style to apply to each level, similar to how it is done in MIS Builder. The style would define the font, background color, padding, etc..

feketemihai replied:
@jbeficent I agree with the style, but probably will be easier to have a dedicated module for styling on reporting-engine, so it can be inheritable...

feketemihai replied:
@jbeficent @OCA/mis-builder-maintainers @OCA/reporting-engine-maintainers I can try to separate the style from mis_builder module into a separate module in reporting-engine and in mis_builder and  account_financial_report to use the style from newly created module. What do you think?

feketemihai replied:
@jbeficent I suggest to leave it like this, until we will have a decision.

rgarnau commented:
Tested on runbot :+1:

rgarnau reviewed (approved):
Tested on runbot :+1:

feketemihai commented:
I will add tests for hierarchy_on on monday...

JordiBForgeFlow commented:
It does not work wit l10n_es. I think that you are doing some tricks with code_prefix... Are they really needed?

feketemihai commented:
@jbeficent When you choose the hierarchy on compute or on childs? Compute should take account that start with code_prefix, childs takes direct accounts linked with the account.group

feketemihai commented:
@jbeficent I think the problem is from l10n_es, the group should act as the view accounts, code_prefix is the code from view account, and name is the name of the account, in l10n_es it's put the code_prefix as name...

feketemihai commented:
Please see l10n_ro account group https://github.com/odoo/odoo/pull/22105/files#diff-0dc5f7b8a9c55073ee43f00597c76514 versus spain https://github.com/odoo/odoo/blob/11.0/addons/l10n_es/data/account_group.xml
The idea is if i take the name of the group, you will only have 103..

feketemihai commented:
@jbeficent Updated code to take name if code_prefix is not set, for compute and in report...

JordiBForgeFlow commented:
Thanks! We will send a PR to odoo to update the account groups in l10n_es. Cc @pedrobaeza

feketemihai commented:
@jbeficent can you check also the vat report?

feketemihai commented:
@jbeficent Until then, i updated to take value from name if code_prefix is empty, but as a design it will be much better with code_prefix and name different.

feketemihai commented:
@jbeficent @rgarnau @pedrobaeza @Yajo  Could you help me with little a thing, i modified the widget to have link to mulri record( e.g. account_move_line) that is launched from the amounts, after i changed the widget of the amount field to monetary, it seems that only one string is linked with the action (depending on country format, could be the currency or the amount - second i think in the json that is contructed) and for the first one i have the error:
"<function Home.web_client at 0x7fae14f9fe18>, /web: Function declared as capable of handling request of type 'http' but called with a request of type 'json'"
How i can include all the string from monetary field to be linkable with the action?

pedrobaeza commented:
I think it's a problem of the widget itself and you are not going to being able to fix it without fixing it upstream.

feketemihai commented:
@jholze I need you designing ideas to improve the styling of the reports

jholze commented:
@feketemihai i send already fo Ahmed but we are still very stressed because of a migration. But yes, would be nice to improve the layout for it.

feketemihai commented:
@jbeficent @rgarnau @mreficent @etobella @pedrobaeza @jholze I think i finish with this, waiting for reviews, style can be improved later.

JordiBForgeFlow reviewed (approved):
Great improvements!

JordiBForgeFlow commented:
Can you squash your commits?

feketemihai commented:
Yep, waiting for comments in case need to change something..

elicoidal reviewed (approved):
LGTM: impressive work
Functional tests OK

feketemihai commented:
@elicoidal It was, i hope i will see a lot of PR on Odoo with Localization update.

jholze commented:
👍
